### DISCUSSION TITLE
Can't Connect From AWS

### DISCUSSION THREAD

**Discussion by fletc3her** [community/community#23781] [New to GitHub]
<p>I have a virtual machine hosted at AWS which I’ve been using GitHub on for some years but this morning all git commands like push/pull which rely on connecting to GitHub are failing. My last commit was five days ago and none of the settings have changed.</p>
<p>Both of these commands time out from the AWS hosted machines but work fine from my local machine.</p>
<pre><code class="lang-auto">ssh -T git@github.com
curl https://github.com/ -I
</code></pre>
<p>I’m not sure where to go from here in terms of troubleshooting. I can access other remote machines fine through either ssh or curl so it doesn’t seem to be a general network problem but rather GitHub specific.</p>
<pre><code class="lang-auto"># git pull
ssh: connect to host github.com port 22: Connection timed out
fatal: The remote end hung up unexpectedly
</code></pre>

**Comment by fletc3her** [comment]
<p>I have contacted support as well.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/fletc3her">@fletc3her</a>, Welcome to the community! I can confirm we’ve received your support request and  a member of our support team will contact you. In the meanwhile would you check your /etc/hosts file on the AWS hosted machines, please? We’ve recently rotated some IP addresses out of our ranges and that could be the reason you are unable to connect.</p>
<p>You can test your connection at this link: <a href="https://github-debug.com/">https://github-debug.com/</a></p>

**Comment by fletc3her** [comment]
<p>Thank you. It was an old address hard-coded in /etc/hosts.</p>
<p>I think this was an old practice to prevent cluttering the known hosts and prompting for IP addresses to be added when performing git actions.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Awesome, I’m glad you could get it going :smile:</p>
